pkg/translator/prometheusremotewrite/metrics_to_prw.go

```diff
@@ -25,9 +27,37 @@ type Settings struct {
 }
 
 // FromMetrics converts pmetric.Metrics to Prometheus remote write format.
-func FromMetrics(md pmetric.Metrics, settings Settings) (tsMap map[string]*prompb.TimeSeries, errs error) {
-	tsMap = make(map[string]*prompb.TimeSeries)
+//
+// Deprecated: FromMetrics exists for historical compatibility and should not be used.
+// Use instead PrometheusConverter.FromMetrics() and then PrometheusConverter.TimeSeries()
+// to obtain the Prometheus remote write format metrics.
+func FromMetrics(md pmetric.Metrics, settings Settings) (map[string]*prompb.TimeSeries, error) {
+	c := NewPrometheusConverter()
+	errs := c.FromMetrics(md, settings)
+	tss := c.TimeSeries()
+	out := make(map[string]*prompb.TimeSeries, len(tss))
+	for i := range tss {
+		out[strconv.Itoa(i)] = &tss[i]
+	}
+
+	return out, errs
+}
```

Comment:
This is not covered by benchmarks or unit tests and I believe it should be.

Reply:
I converted `BenchmarkFromMetrics` to use `FromMetrics`, and made a new one for `PrometheusConverter`: `BenchmarkPrometheusConverter_FromMetrics`. Is this enough? I notice that the existing test suite already doesn't call `FromMetrics`.